Eslint rule failure cleanup

zacbraddy opened the pull request:
I noticed there were a lot of warnings and a few failures pertaining to slight formatting issues. I thought I'd clear them up as it twinged my OCD :D 

Most warnings were line length problems, I ignored lines that were lengthy because of string values or if they were the definition of a function. For any function calls I just put each parameter on a new line.

Also got rid of a FIXME related to eslint that didn't appear to be a problem anymore.

Might want to consider modifying the eslint rules to ignore string literals and comments to avoid more of these?

zacbraddy commented:
It appears that for the errors I've suffered from a bit of "Works on my machine" syndrome or rather "Doesn't work on my machine syndrome" when I run an npm run lint locally (I'm on a windows box) I get linting errors saying that those lines I reverted in the second commit are two spaces too far to the right, thus the PR.

I'll investigate further locally to try and work out what is happening with those errors and report what I find, in the mean time at least the linter is warning free now :)

bvaughn commented on src/renderers/dom/fiber/ReactDOMFiber.js:
```diff
@@ -325,7 +325,9 @@ function warnAboutUnstableUse() {
 
+/* eslint-disable max-len */
 function renderSubtreeIntoContainer(parentComponent : ?ReactComponent<any, any, any>, children : ReactNodeList, containerNode : DOMContainerElement | Document, callback: ?Function) {
+/* eslint-enable max-len */
```
Rather than bypassing the lint check, seems better to split params across lines:

```js
function renderSubtreeIntoContainer(
  parentComponent : ?ReactComponent<any, any, any>,
  children : ReactNodeList, containerNode : DOMContainerElement | Document,
  callback: ?Function
) {
```

This is what we normally do (eg [`createInstance `](https://github.com/facebook/react/blob/3c75593692524678fbb74e8077018cc39b04ffaa/src/renderers/dom/fiber/ReactDOMFiber.js#L171-L177) above).

bvaughn commented on src/renderers/dom/fiber/ReactDOMFiber.js:
```diff
@@ -355,7 +357,10 @@ var ReactDOM = {
+  unstable_renderSubtreeIntoContainer(parentComponent : ReactComponent<any, any, any>,
+                                      element : ReactElement<any>,
+                                      containerNode : DOMContainerElement | Document,
+                                      callback: ?Function) {
```
This isn't how we normally indent params. Check out [`createInstance `](https://github.com/facebook/react/blob/3c75593692524678fbb74e8077018cc39b04ffaa/src/renderers/dom/fiber/ReactDOMFiber.js#L171-L177) or [`prepareUpdate`](https://github.com/facebook/react/blob/3c75593692524678fbb74e8077018cc39b04ffaa/src/renderers/dom/fiber/ReactDOMFiber.js#L214-L221) above for examples.

bvaughn commented on scripts/release-manager/commands/stable-prs.js:
```diff
@@ -131,7 +131,9 @@ module.exports = function(vorpal, app) {
                 if (!pr.merged_at) {
+                  /* eslint-disable max-len */
                   this.log(`${chalk.yellow.bold('WARNING')} ${pr.html_url} was not merged, should have the milestone unset.`);
+                  /* eslint-enable max-len */
```
nit: Same as [above](https://github.com/facebook/react/pull/8846/files#r98274532).

bvaughn commented on scripts/release-manager/commands/stable-prs.js:
```diff
@@ -168,7 +170,9 @@ module.exports = function(vorpal, app) {
               .catch((err) => {
+                /* eslint-disable max-len */
                 this.log(`${chalk.red.bold('ERROR')} Something went wrong and your repo is probably in a bad state. Sorry.`);
+                /* eslint-enable max-len */
```
nit: Same as [above](https://github.com/facebook/react/pull/8846/files#r98274532).

bvaughn commented on scripts/release-manager/cli.js:
```diff
@@ -98,7 +98,9 @@ const app = {
     } catch (e) {
+      /* eslint-disable max-len */
       console.error('Attempt to load config file failed. Please run `init` command for initial setup or make sure ~/.react-release-manager.json is valid JSON. Using a default config which may not work properly.');
+      /* eslint-enable max-len */
```
nit: Probably better to split string across lines rather than bypass lint rule (eg like [here](https://github.com/facebook/react/blob/3c75593692524678fbb74e8077018cc39b04ffaa/src/renderers/dom/fiber/ReactDOMFiber.js#L324-L328)).

bvaughn commented on src/renderers/shared/fiber/ReactFiber.js:
```diff
@@ -331,7 +331,9 @@ exports.createFiberFromText = function(content : string, priorityLevel : Priorit
 
+/* eslint-disable max-len */
 function createFiberFromElementType(type : mixed, key : null | string, debugOwner : null | Fiber | ReactInstance) : Fiber {
+/* eslint-enable max-len */
```
Same as [above](https://github.com/facebook/react/pull/8846/files#r98273555).

bvaughn commented on src/renderers/shared/fiber/ReactFiberCompleteWork.js:
```diff
@@ -201,7 +201,15 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
+              oldProps,
+              newProps,
+              rootContainerInstance,
+              currentHostContext);
```
nit: `);` on new line is more readable (eg see [here](https://github.com/facebook/react/blob/3c75593692524678fbb74e8077018cc39b04ffaa/src/renderers/shared/fiber/ReactFiberCompleteWork.js#L243))

bvaughn commented on src/renderers/shared/fiber/ReactFiberReconciler.js:
```diff
@@ -49,18 +49,35 @@ export type HostConfig<T, P, I, TI, PI, C, CX, PL> = {
+                 props : P,
+                 rootContainerInstance : C,
+                 hostContext : CX,
+                 internalInstanceHandle : OpaqueNode) : I,
```
Same as [here](https://github.com/facebook/react/pull/8846/files#r98273982) for various methods in this file.

bvaughn commented on src/renderers/shared/fiber/ReactFiberReconciler.js:
```diff
@@ -102,7 +118,9 @@ getContextForSubtree._injectFiber(function(fiber : Fiber) {
 
+/* eslint-disable max-len */
 module.exports = function<T, P, I, TI, PI, C, CX, PL>(config : HostConfig<T, P, I, TI, PI, C, CX, PL>) : Reconciler<C, I, TI> {
+/* eslint-enable max-len */
```
Probably better as:
```js
module.exports = function<T, P, I, TI, PI, C, CX, PL>(
  config : HostConfig<T, P, I, TI, PI, C, CX, PL>
) : Reconciler<C, I, TI> {
```

zacbraddy commented:
@bvaughn I don't feel like any of your suggestions were nit picky at all. Thank you so much for taking the time to not only point them out but find me examples of how it's done in the React code base. That really was so useful for helping me to learn the way that things get done stylistically when working with React. All of your suggestions made perfect sense.

I have gone ahead and implemented them all and checked them in.

One other thing, you can see that my original check in was to attempt to fix some linting errors I was getting on my (windows) machine in eslint which is what sparked me doing all this. Unfortunately the same linting errors weren't causing problems on the CI server so my build popped and I reverted them thinking it must have just been my machine. 

I've now researched this a bit and it appears to be something that has gotten reported sporadically over some time in eslint by windows users. The problem specifically was when it comes to linting ternary operators with nested grouped elements like arrays or objects. [Here's a bug](https://github.com/eslint/eslint/issues/6606) with a good explanation of the problem. This bug is closed but I've found many bugs which are variations on the same theme. 

Anyway I have made a change to the two files in the  src/renderers/shared/shared/__tests__/ folder which were giving me the errors and now it appears both the linter on my machine **and** the CI server are happy. The only question that remains is.....are you happy? :)

Cheers again for all the great pointers
Zac

bvaughn commented on scripts/release-manager/commands/stable-prs.js:
```diff
@@ -131,7 +131,10 @@ module.exports = function(vorpal, app) {
-                  this.log(`${chalk.yellow.bold('WARNING')} ${pr.html_url} was not merged, should have the milestone unset.`);
+                  this.log(
+                    `${chalk.yellow.bold('WARNING')} ${pr.html_url} was not merged, 
+                     should have the milestone unset.`
```
This changes the logged text (adding empty space and line breaks in between "merged," and "should"). This isn't desirable. Can you split this text like you did above (in `scripts/release-manager/cli.js`)?

bvaughn commented on scripts/release-manager/cli.js:
```diff
@@ -98,7 +98,11 @@ const app = {
+        'Attempt to load config file failed. Please run `init` command for initial setup or make sure ' +
+        '~/.react-release-manager.json is valid JSON. Using a default config which may not work ' +
+        'properly.'
+      );
```
👍

bvaughn commented on scripts/release-manager/commands/stable-prs.js:
```diff
@@ -168,7 +171,10 @@ module.exports = function(vorpal, app) {
+                this.log(
+                  `${chalk.red.bold('ERROR')} Something went wrong and your repo is
+                   probably in a bad state. Sorry.`
+                );
```
Same comment as above! 😄

zacbraddy commented:
Whoops, sorry about that. I've gone ahead and fixed those mistakes and then squashed up all my commits for you. All good to go provided there isn't anything further?
